### PR TITLE
use different name for ubuntu package

### PR DESCRIPTION
**Pull request by hlawatschek**
in ubuntu and debian based systems, the package name will be different. 
This PR takes the packaging guidelines for ubuntu packages into account.

### CODE REVIEW

**Comment by ares** [comment]
is there a package for foreman_scap_client on these platforms?

**Comment by hlawatschek** [comment]
We are currently building packages for different platforms including debian and ubuntu.

**Comment by hlawatschek** [comment]
See https://github.com/theforeman/foreman_scap_client/pull/24

**Comment by ares** [comment]
ok, will they be available in the foreman-packaging repo? just curious if we need some additional if conditions here

File: manifests/init.pp [hunk 1]
```diff
@@ -104,7 +104,18 @@
     }
   }
 
-  package { 'rubygem-foreman_scap_client': ensure => $ensure, install_options => $install_options, } ->
+  $scappackage = $operatingsystem ? {
```
Comment (ekohl) [inline manifests/init.pp]: It's much cleaner to do this in `params.pp`

File: manifests/init.pp [hunk 2]
```diff
@@ -104,7 +104,18 @@
...
+    ubuntu => rubygem-foreman-scap-client,
```
Comment (ekohl) [inline manifests/init.pp]: Debian (and thus Ubuntu) conventions say it should be `ruby-foreman-scap-client`.

File: manifests/init.pp [hunk 3]
```diff
@@ -104,7 +104,18 @@
...
+    default => rubygem-foreman_scap_client
+  }
+
+  package { $scappackage: ensure => $ensure, install_options => $install_options, } ->
```
Comment (ekohl) [inline manifests/init.pp]: Since you're renaming this, it'd be cleaner to also use the variable at line 103 in the `Package`.

**Comment by hlawatschek** [comment]
Probably/hopefully the packages will be available in the foreman-packaging repo ;-)

File: manifests/init.pp [hunk 4]
```diff
@@ -46,6 +46,8 @@
 #                     if it's not array, it's automatically converted to it, so you can
 #                     even specify just one policy as a hash
 #                     type:array
+#                     
+# $ruby_foreman_scap_client_package:: os dependent package name for rubygem-foreman_scap_client package
```
Comment (ekohl) [inline manifests/init.pp]: This implies it's a class parameter, but it's not so you either can remove this or add it as a parameter.

File: manifests/params.pp [hunk 5]
```diff
@@ -31,6 +31,11 @@
     $rh_certificate_consumer_host_key = undef
   }
 
+  $ruby_foreman_scap_client_package = $operatingsystem ? {
```
Comment (ekohl) [inline manifests/params.pp]: You could simplify this to just `$package_name`. You can also use `$::osfamily` where `Debian` covers both Debian and Ubuntu (and their derivatives even).

File: manifests/params.pp [hunk 6]
```diff
@@ -31,6 +31,11 @@
...
+    /Debian|Ubuntu/ => ruby-foreman-scap-client,
```
Comment (ekohl) [inline manifests/params.pp]: Please use quotes in strings.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 4 above)
Comment (hlawatschek) [reply manifests/init.pp]: removed.

File: manifests/params.pp (hunk 5 above)
Comment (hlawatschek) [reply manifests/params.pp]: using $package_name

File: manifests/params.pp (hunk 6 above)
Comment (hlawatschek) [reply manifests/params.pp]: added $osfamily and added quotes.
Thanks a lot for reviewing my changes.

**Comment by hlawatschek** [comment]
Hi! Is there anything else that I should do for this PR?

File: manifests/init.pp [hunk 7]
```diff
@@ -100,11 +101,17 @@
...
+      before   => Package[$package_name]
     }
   }
 
-  package { 'rubygem-foreman_scap_client': ensure => $ensure, install_options => $install_options, } ->
+  package { $package_name: ensure => $ensure, install_options => $install_options, } ->
+  
+  file { '/etc/foreman_scap_client':
+    ensure => directory,
+    owner   => 'root',
```
Comment (ekohl) [inline manifests/init.pp]: Could you align the `=>`?

File: manifests/params.pp [hunk 8]
```diff
@@ -31,6 +31,11 @@
     $rh_certificate_consumer_host_key = undef
   }
 
+  $package_name = $osfamily ? {
+    'Debian' => 'ruby-foreman-scap-client',
+    default => 'rubygem-foreman_scap_client'
```
Comment (ekohl) [inline manifests/params.pp]: Could you align the `=>`?

File: manifests/params.pp [hunk 9]
```diff
@@ -31,6 +31,11 @@
     $rh_certificate_consumer_host_key = undef
   }
 
+  $package_name = $osfamily ? {
```
Comment (ekohl) [inline manifests/params.pp]: It's more common to write `$::osfamily` to make sure you're using the fact and would be consistent with the rest of the file.

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by ekohl** [comment]
In https://github.com/theforeman/puppet-foreman_scap_client/pull/49 I've added the basics to test the module. It would have told you some of those linting errors.

It'd be nice if the distros were also listed in metadata.json.

File: manifests/init.pp (hunk 7 above)
Comment (hlawatschek) [reply manifests/init.pp]: OK :-)

File: manifests/params.pp (hunk 9 above)
Comment (hlawatschek) [reply manifests/params.pp]: OK :-)

**Comment by hlawatschek** [comment]
Thanks for reviewing my changes and being so patient with me :-)
puppet-lint seems to be quite happy now :-)

File: manifests/init.pp [hunk 10]
```diff
@@ -65,7 +66,7 @@
   $cron_template          = 'foreman_scap_client/cron.erb',
   $cron_splay             = 600,
 ) inherits foreman_scap_client::params {
-
+  $package_name = $::foreman_scap_client::params::package_name
```
Comment (ekohl) [inline manifests/init.pp]: It'd be also fine to put this in as an actual parameter

**Review by ekohl — APPROVED** [review]
The code looks good. Could you squash the changes? If not, I think we can also do it on merge.

**Comment by hlawatschek** [comment]
I did a rebase and squashed all commits together. 
Also I added $package_name as actual parameter.

**Review by ekohl — APPROVED** [review]

**Comment by ares** [comment]
Thanks @hlawatschek, meging!

**Pull request merged**